### DISCUSSION TITLE
`"workspace:*"` vs `"*"` in `package.json`'s `dependencies`

### DISCUSSION THREAD

**Discussion by yamcodes** [vercel/turborepo#7687] [Help] [answered]
### Summary

1. What is the difference between listing `"workspace:*"` and `"*"` in the version field of a cross-referenced package?
2. Is this notation reserved for internal packages, or is it applicable for external packages as well?

**Comment by anthonyshew** [comment, marked as answer]
`workspace:*` is a protocol specific to pnpm and later versions of yarn to reference Internal Packages. It will never reach out to npm for a package with the name being asked for.

`*` means "any version" and can reference either External Dependencies or Internal Packages in npm and early yarn. This makes it's usage somewhat murky so your confusion is understandable.
